boot/zephyr/CMakeLists.txt

```diff
@@ -168,3 +171,12 @@ if (NEED_TINYCRYPT)
   target_sources(app PRIVATE "${TINYCRYPT_DIR}/source/sha256.c")
   target_sources(app PRIVATE "${TINYCRYPT_DIR}/source/utils.c")
 endif()
+
+if (CONFIG_MCUBOOT_SERIAL)
+zephyr_sources(${BOOT_DIR}/zephyr/serial_adapter.c)
+
+add_subdirectory(${BOOT_DIR}/boot_serial ./boot/boot_serilal)
```

Comment:
This haw I described modules for zephyr build in these many files is dictating by zephyr approach to this: one module/directory - one CMakeLists.txt For sure this is not requirement for mcuboot so it could be reworked later.